### PR TITLE
chore: first pass at dir restructure

### PR DESCRIPTION
**Pull request by duglin**
I think the README needs more work to talk about the various specs but this at least this moves things into the new locations.

Signed-off-by: Doug Davis <dug@us.ibm.com>

### CODE REVIEW

**Comment by duglin** [comment]
The dir structure becomes this:
```
./schemaregistry
./share
./subscriptions
./discovery
./artwork
./pagination
./cesql
./community
./tools
./cloudevents
./cloudevents/extensions
./cloudevents/adapters
```

Should we create:
```
./cloudevents/bindings
./cloudevents/formats
```
too?

**Comment by slinkydeveloper** [comment]
I think you didn't rebased, can you move `cesql_tck` under `cesql/tck`?

File: README.md [hunk 1]
```diff
@@ -49,8 +49,9 @@ The following documents are available:
 | Proprietary Specifications    |                                        -                                        |      [master](https://github.com/cloudevents/spec/blob/master/proprietary-specs.md)      |
```
Comment (slinkydeveloper) [inline README.md:49]: All the links in this file seems broken

**Review by jskeet — COMMENTED** [review]
It looks like I can't add comments to files that are just moved, so...

- Pagination: it's even clearer under this new structure that pagination is a bit of an orphan. I'd expect pagination to be specified by any specific API (e.g. a subscriptions API) rather than being on its own like this. I'd like to talk about this one more.

- It feels like we have several non-spec things... I wonder whether they could either go in a separate repository, or be bundled into a single directory:
  - Share
  - Artwork
  - Tools
  - (.github - but that's pretty much enforced in its location)

At the last meeting we started considering the idea of versioning being applied to bundles of specs, and it looks like the direction of this PR is each top-level directory being a bundle - is that correct? If so, that's really useful for my still-to-be-done work on proposing a specific versioning scheme. As I think you (Doug) have said before, we could potentially *consider* splitting each bundle into its own repo.

**Comment by JemDay** [comment]
+1 to having a separate directory for `bindings` & `formats`, I will admit it feels a bit weird to have a `cloudevents` directory but I can live with that ..

**Comment by duglin** [comment]
closing in favor of:  #893

**Pull request closed without merging**